tjmonopix2/scans/eudaq_producer.py

```diff
@@ -0,0 +1,490 @@
+    'max_triggers': False,
+    # number of maximum received triggers after stopping readout, if False no limit on received trigger
+
+    'trigger_latency': 100,  # latency of trigger in units of 25 ns (BCs)
+    'trigger_delay': 57,  # trigger delay in units of 25 ns (BCs)
+    'trigger_length': 32,  # length of trigger command (amount of consecutive BCs are read out)
+    'veto_length': 3200,
+    # raw val (si val, 1/si val): real hitrate
+    #    400 (  10 us, 100 kHz): 1.5kHz (lim by telescope)
+    #  25000 ( 625 us, 1.6 kHz): 580 Hz
+    #  50000 (1250 us, 0.8 kHz): 300 Hz
+    # 150000 (3750 us, 0.3 kHz): 100 Hz
+    
+    # 
+    # length of TLU veto in units of 25 ns (BCs). This vetos new triggers while not all data is received.
+    # Should be adjusted for longer trigger length.
+
+    # Trigger configuration
+    'bench': {'TLU': {
+        'TRIGGER_MODE': 3,
+        # Selecting trigger mode: Use trigger inputs/trigger select (0), TLU no handshake (1), TLU simple
+        # handshake (2), TLU data handshake (3)
+        'TRIGGER_SELECT': 0,  # Selecting trigger input: HitOR (1), disabled (0)
+        'TRIGGER_HANDSHAKE_ACCEPT_WAIT_CYCLES': 20
+        # TLU trigger minimum length in TLU clock cycles. Change default here in order to ignore glitches from buggy
+        # original TLU FW.
+    }
+    }
+}
+
+
+class EudaqScan(scan_ext_trigger.ExtTriggerScan):
+    scan_id = 'eudaq_scan'
+
+    min_spec_occupancy = False  # needed for handle data function of ext. trigger scan
+
+    def __init__(self, daq_conf=None, bench_config=None, scan_config={}, scan_config_per_chip=None, suffix='', run_number=998):
+        super().__init__(daq_conf, bench_config, scan_config, scan_config_per_chip, suffix)
+        self.last_readout_data = None
+        self.last_trigger = 0
+        self.bdaq_recording = True
+        self.ovflw_cnt = 0
+        self.run_number = run_number
+
+    def __del__(self):
+        pass
+        # print('destructor')
+        # self.close()
+
+    def _configure(self, callback=None, **_):
+        super(EudaqScan, self)._configure(**_)
+
+        self.last_readout_data = np.array([], dtype=np.uint32)
+
+    def handle_data(self, data_tuple, receiver=None):
+
+        #print('handle data')
+
+        if self.bdaq_recording:
+            super(EudaqScan, self).handle_data(data_tuple, receiver=None)
+
+        raw_data = data_tuple[0]
+
+        if np.any(self.last_readout_data):  # no last readout data for first readout
+            actual_data = np.concatenate((self.last_readout_data, raw_data))
+        else:
+            actual_data = raw_data
+
+        trg_idx = np.where(actual_data & au.TRIGGER_HEADER > 0)[0]
+        trigger_data = np.split(actual_data, trg_idx)
+
+        # Send data of each trigger
+        for i  in range(0, len(trigger_data) - 2):
+            dat = trigger_data[i]
+            glitch_detected = False
```

Comment:
What kind of glitch is this? I never observed a glitching trigger number that is bit-shifted by 1?